v2/handler.js

```diff
@@ -966,3 +973,25 @@ function InRequestOptions(
     self.hostPort = hostPort;
     self.tracer = tracer;
 }
+
+function validateHostPort(hostPort) {
+    var parts = hostPort.split(':');
+    if (parts.length !== 2) {
+        return false;
+    }
+
+    if (parts[0].split('.').length !== 4) {
```

Comment:
doesn't validate that the first part is a dotted quad, we need to disallow dns names; should also disallow 0.0.0.0